app/models/manageiq/providers/vmware/infra_manager/orchestration_template.rb

```diff
@@ -1,4 +1,65 @@
-class ManageIQ::Providers::Vmware::InfraManager::OrchestrationTemplate < OrchestrationTemplate
+class ManageIQ::Providers::Vmware::InfraManager::OrchestrationTemplate < ::OrchestrationTemplate
+  belongs_to :ext_management_system, :foreign_key => "ems_id", :class_name => "ManageIQ::Providers::Vmware::InfraManager", :inverse_of => false
+
+  SPEC_KEY_MAPPING = {
+    "resource_pool" => "resource_pool_id",
+    "ems_folder"    => "folder_id",
+    "host"          => "host_id"
+  }.freeze
+
+  def connect
+    require 'vsphere-automation-cis'
+
+    configuration = VSphereAutomation::Configuration.new.tap do |c|
+      c.host = ext_management_system.hostname
+      c.username = ext_management_system.auth_user_pwd.first
+      c.password = ext_management_system.auth_user_pwd.last
+      c.verify_ssl = false
+      c.verify_ssl_host = false
+    end
+
+    api_client = VSphereAutomation::ApiClient.new(configuration)
+    VSphereAutomation::CIS::SessionApi.new(api_client).create('')
+    api_client
+  end
```

Comment:
Future refactoring, I'd like the connection method for this vsphere-automation sdk to live with the InfraManager since we also use this in refresh [here](https://github.com/ManageIQ/manageiq-providers-vmware/blob/master/app/models/manageiq/providers/vmware/infra_manager/inventory/collector.rb#L272-L284)

We could pass in a `:service` similar to what we do with e.g. AWS connections to EC2/RDS/S3